docs/standards.md

```diff
@@ -0,0 +1,145 @@
+
+   - We merge using the squash+rebase strategy in GitHub. The GitHub behavior is
+     something like this:
+
+     ```sh
+     git checkout main
+     git merge --squash <branch>
+     git push
+     ```
+
+     > [!NOTE]  
+     > You cannot actually do it manually in our repo as the `main` branch does
+     > not allow remote pushes.
+
+```mermaid
+%%{ init: { "flowchart": { "curve": "linear" }}}%%
+graph TD;
+  A[Open pull<br>request]-->B{Ready to<br>review?};
+  B--no-->A
+  B--yes-->C{Do you know<br>who reviews?}
+  C--no-->D[Assign leads<br>to review]
+  D-->E[Assign reviewers]
+  C--yes-->E
+  E-->F{Reviewers<br>approved?}
+  F--no-->G[Make changes]-->B
+  F--yes-->H[Merge]
+```
+
+#### Branch protections
+
+PRs cannot be merged into `main` unless they satisfy our branch protection
+rules:
+
+- All PRs must be reviewed and approved by at least one other person.
+- All PRs must pass all required automated checks. The list of required checks
+  will evolved over time.
+- Any code that is exempted from automated tests must be documented and
+  explained in code itself as well as the pull request. Code my be exempted due
+  to false positives that cannot be resolved in other ways; true positives that
+  will be resolved in a future fix; or similar scenarios.
+- All commits in a PR must be
+  [cryptographically signed](https://calebhearth.com/sign-git-with-ssh)
+  by their authors.
+  ([GPG signatures](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits)
+  are also allowed, though SSH is easier to configure and thus encouraged.)
+- All merges into `main` will be pushed into a merge queue. Merge queues help
+  reduce the friction of merging PRs where multiple need to be done at once. PRs
+  go into the queue and get batched up, tested, and merged all at once. There's
+  no need for PR authors to merge from `main` before merging _into_ `main`.
+
+### Merging strategy
+
+We use squash merging in the merge queue. PR authors do not need to select
+anything as this is handled automatically.
+
+### Deployment strategy
+
+Deployments into preview and/or staging environments happen automatically from
+the appropriate branches.
+
+Deployments into production will be managed via tagged releases. When a new tag
+is created in the form `v1.3`, a new production deployment will be triggered
+from that tag. We will create tags via GitHub releases rather than creating them
+manually so we can associate them with release notes, documenting what is
+changing. Using tags also allows us an opportunity to review an entire release
+candidate as a whole before deploying it.
+
+> [!WARNING]  
+> Even though pushing to the `main` branch is disabled, tags applied to commits
+> on the `main` branch **_can_** be pushed. Be very careful with manually
+> tagging the repo. If your tag matches the `vX.Y` format indicating a new
+> production release, it **_will_** trigger a deployment into production.
```

Comment:
Is there a way to remind us of this? maybe in a PR checklist?

Reply:
Not really. I expect manual tagging to be rare anyway, and even when we do it, I think we're almost certain to never accidentally run into this scenario. But tags are not associated with pull requests or anything else, so a reminder in a PR wouldn't really be helpful. Tags are mostly just a human-readable reference to a specific commit in the git history, so they're out of cycle from all the rest of our work.

This particular section is also just an idea of how we might trigger deploys so we've got a place in our documentation for it. We'll fill it in with real information soon and it's possible we won't use tags at all. As long as we are using GitHub Actions to run our deployments, I'd argue we should trigger deploys from releases, not tags, as we have more control and auditability over those events.